dopamine/experiment5/experiment5_varplots.jl

```diff
@@ -0,0 +1,176 @@
+# VARIANCE IN SPEED PLOTS
+# Plots of wild type and mutant on bacterial lawn (Omura)
+
+using DataFrames
+using CSV
+using CategoricalArrays
+using GLMakie
+using Statistics
+using StatsBase
+
+
+
+# LOAD SPEEDS CSV INTO DATAFRAME
+experimentdir = @__DIR__
+
+speeds = DataFrame(CSV.File(joinpath(experimentdir, "speeds.csv")))
+
+# CALCULATE MEAN AND STD SPEED OF EACH TRACK
+tracks = groupby(speeds, [:experiment, :id, :track])
+trackstats = combine(tracks, :speed => mean => :meanspeed, :speed => std => :stdspeed)
+
+# FILTER TRACKS WITH NAN VALUES FOR STD
+filter!(:stdspeed => x -> !(isnan(x)), trackstats)
+
+# BIN TRACKS ACCORDING TO MEAN SPEED
+bin = Vector{Int}()
+
+for row in 1:nrow(trackstats)
+    if 0 ≤ trackstats.meanspeed[row] < 50
+        push!(bin, 1)
+    elseif 50 ≤ trackstats.meanspeed[row] < 100
+        push!(bin, 2)
+    elseif 100 ≤ trackstats.meanspeed[row] < 150
+        push!(bin, 3)
+    elseif 150 ≤ trackstats.meanspeed[row] < 200
+        push!(bin, 4)
+    elseif 200 ≤ trackstats.meanspeed[row] < 250
+        push!(bin, 5)
+    elseif 250 ≤ trackstats.meanspeed[row] < 300
+        push!(bin, 6)
+    elseif 300 ≤ trackstats.meanspeed[row]
+        push!(bin, 7)
+    end
+end
+
+trackstats.bin = bin
+
+# MAKE CATEGORICAL ARRAYS FOR PLOTTING
+trackstats.medium = categorical(map(i-> split(i, '_')[1], trackstats.id), levels = ["M9", "DA"])
+trackstats.worm = categorical(map(i-> split(i, '_')[2], trackstats.id), levels = ["N2", "CB", "MT"])
+trackstats.bacteria = categorical(map(i-> split(i, '_')[3], trackstats.id), levels = ["NGM", "OP50"])
+
+# ONLY KEEP DATA ON BACTERIAL LAWN
+filter!(:bacteria => x -> x == "OP50", trackstats)
+
+# SEPARATE ALL CONDITIONS INTO NEW DFs
+buffer = filter(:medium => x -> x == "M9", trackstats)
+bufferN2 = filter(:worm => x -> x == "N2", buffer)
+bufferCB = filter(:worm => x -> x == "CB", buffer)
+bufferMT = filter(:worm => x -> x == "MT", buffer)
+
+dopamine = filter(:medium => x -> x == "DA", trackstats)
+dopamineN2 = filter(:worm => x -> x == "N2", dopamine)
+dopamineCB = filter(:worm => x -> x == "CB", dopamine)
+dopamineMT = filter(:worm => x -> x == "MT", dopamine)
+
+# GET MEAN AND STD OF MEAN SPEED AND SEM OF EACH BIN OF EACH CONDITION
+# x = mean of mean speed
+# y = mean of std speed
+# error = sem of std speed
+bufferN2binned = combine(groupby(bufferN2, [:bin]), :meanspeed => mean => :meanofmeanspeed, :stdspeed => mean => :meanofstdspeed, :stdspeed => sem => :semofstdspeed, :track => length => :n)
```

Comment:
```suggestion
bufferN2binned = combine(groupby(bufferN2, [:bin]), 
            :meanspeed => mean => :meanofmeanspeed,
            :stdspeed  => mean => :meanofstdspeed,
            :stdspeed  => sem  => :semofstdspeed,
            :track => length => :n)
```

just for readability